### PR TITLE
add store labels

### PR DESCRIPTION
**Pull request by L3T**
### What problem does this PR solve?

Fix https://github.com/pingcap/tidb-operator/issues/3783

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support setting customized store labels according to the node labels
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -468,6 +468,10 @@ type TiKVSpec struct {
 	// StorageVolumes configure additional storage for TiKV pods.
 	// +optional
 	StorageVolumes []StorageVolume `json:"storageVolumes,omitempty"`
+
+	// StoreLabels configure additional labels for TiKV stores.
+	// +optional
+	StoreLabels []string `json:"StoreLabels,omitempty"`
```
Comment (lonng) [inline pkg/apis/pingcap/v1alpha1/types.go:474]:
```suggestion
	StoreLabels []string `json:"storeLabels,omitempty"`
```

File: pkg/manager/member/tikv_member_manager.go [hunk 2]
```diff
@@ -816,8 +816,8 @@ func (m *tikvMemberManager) setStoreLabelsForTiKV(tc *v1alpha1.TidbCluster) (int
 		return setCount, err
 	}
 
-	locationLabels := []string(config.Replication.LocationLabels)
-	if locationLabels == nil {
+	StoreLabels := append(config.Replication.LocationLabels, tc.Spec.TiKV.StoreLabels...)
```
Comment (lonng) [inline pkg/manager/member/tikv_member_manager.go:819]:
```suggestion
	storeLabels := append(config.Replication.LocationLabels, tc.Spec.TiKV.StoreLabels...)
```

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 3]
```diff
@@ -468,6 +468,10 @@ type TiKVSpec struct {
 	// StorageVolumes configure additional storage for TiKV pods.
 	// +optional
 	StorageVolumes []StorageVolume `json:"storageVolumes,omitempty"`
+
+	// StoreLabels configure additional labels for TiKV stores.
```
Comment (lonng) [inline pkg/apis/pingcap/v1alpha1/types.go:472]:
```suggestion
	// StoreLabels configures additional labels for TiKV stores.
```

**Comment by DanielZhangQD** [comment]
@L3T Could you please fix the CI and update the test result in the PR description?

**Comment by DanielZhangQD** [comment]
@L3T Please fix the CI.

**Comment by L3T** [comment]
手动进行测试，更新 crd 以及 operator 版本到本分支
为 tikv 添加        
storeLabels:
      - testLabel
 观察 operator 日志 
image
为 tikv-0 tikv-1 所在节点添加 testLabel:x label
观察operator 日志 
image

**Comment by DanielZhangQD** [comment]
@L3T There is not location labels in the test result, could you please verify the case including the location labels and customized labels?

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: pkg/manager/member/tikv_member_manager.go [hunk 4]
```diff
@@ -816,8 +816,8 @@ func (m *tikvMemberManager) setStoreLabelsForTiKV(tc *v1alpha1.TidbCluster) (int
 		return setCount, err
 	}
 
-	locationLabels := []string(config.Replication.LocationLabels)
-	if locationLabels == nil {
+	storeLabels := append(config.Replication.LocationLabels, tc.Spec.TiKV.StoreLabels...)
```
Comment (lonng) [inline pkg/manager/member/tikv_member_manager.go:819]: What if there are some duplicated labels?

**Comment by L3T** [comment]
添加 host location_labels 
image
观察 store label 
image
能争取设置 store labels 和 location labels

File: pkg/manager/member/tikv_member_manager.go (hunk 4 above)
Comment (L3T) [reply pkg/manager/member/tikv_member_manager.go:819]: 这里重复的话，下面逻辑都是根据 storeLabels 对 map 赋值，所以对下面逻辑不会有影响，改了还要多几行代码(，这里就不改了吧(

**Pull request merged**